src/middlewared/middlewared/plugins/vm.py

```diff
@@ -929,13 +929,26 @@ def random_mac(self):
         else:
             return update_devices
 
-    @accepts(Int('id'))
-    async def do_delete(self, id):
+    @accepts(Int('id'), Bool('zvols', default=False))
```

Comment:
I would turn this into a Dict with zvols inside, so we can easily add new options later.